src/components/tl/mlx5/mcast/p2p/ucc_tl_mlx5_mcast_p2p.c

```diff
@@ -0,0 +1,148 @@
+    if (ucc_unlikely(UCC_OK != status)) {
+        tl_error(ctx->lib, "nonblocking p2p init failed");
+        return status;
+    }
+
+    ((ucc_tl_mlx5_mcast_p2p_completion_obj_t *)args.cb.data)->req = req;
+
+    status = ucc_collective_post(req);
+    if (ucc_unlikely(UCC_OK != status)) {
+        tl_error(ctx->lib, "nonblocking p2p post failed");
+        return status;
+    }
+
+    *p2p_req = req;
+
+    return status;
+}
+
+static inline ucc_status_t do_send_nb(void *sbuf, size_t len, ucc_rank_t my_team_rank,
+                                      int dest, ucc_team_h team, ucc_context_h ctx,
+                                      ucc_coll_callback_t *callback, ucc_coll_req_h *req)
+{
+    return ucc_tl_mlx5_mcast_do_p2p_bcast_nb(sbuf, len, my_team_rank, dest,
+                                             team, ctx, callback, req, 1);
+}
+
+static inline ucc_status_t do_recv_nb(void *rbuf, size_t len, ucc_rank_t my_team_rank,
+                                      int dest, ucc_team_h team, ucc_context_h ctx,
+                                      ucc_coll_callback_t *callback, ucc_coll_req_h *req)
+{
+    return ucc_tl_mlx5_mcast_do_p2p_bcast_nb(rbuf, len, my_team_rank, dest,
+                                             team, ctx, callback, req, 0);
+}
+
+ucc_status_t ucc_tl_mlx5_mcast_p2p_progress(void *context)
+{
+    ucc_tl_mlx5_mcast_oob_p2p_context_t *oob_p2p_ctx;
+    ucc_context_h                        ctx ;
+    ucc_status_t                         status;
+
+    oob_p2p_ctx = (ucc_tl_mlx5_mcast_oob_p2p_context_t *)context;
+    ctx         = oob_p2p_ctx->base_ctx;
+
+    status = ucc_context_progress(ctx);
+    if (ucc_unlikely(UCC_OK != status)) {
+        tl_error(ctx->lib, "p2p progress failed");
+        return status;
+    }
+
+    return status;
+}
+
+ucc_status_t ucc_tl_mlx5_mcast_p2p_send_nb(void* src, size_t size, int rank, int tag,
+                                           void *context, ucc_tl_mlx5_mcast_p2p_completion_obj_t *obj)
+{
+    ucc_tl_mlx5_mcast_oob_p2p_context_t *oob_p2p_ctx  = (ucc_tl_mlx5_mcast_oob_p2p_context_t *)context;
+    ucc_status_t                         status       = UCC_OK;
+    ucc_coll_req_h                       req          = NULL;
+    ucc_rank_t                           my_team_rank = oob_p2p_ctx->my_team_rank;
+    ucc_team_h                           team         = oob_p2p_ctx->base_team;
+    ucc_context_h                        ctx          = oob_p2p_ctx->base_ctx;
+    ucc_coll_callback_t                  callback;
+
+    callback.cb   = ucc_tl_mlx5_mcast_completion_cb;
+    callback.data = obj;
+
+    status = do_send_nb(src, size, my_team_rank, rank, team, ctx, &callback, &req);
+
+    if (status != UCC_INPROGRESS && status != UCC_OK) {
+        tl_error(ctx->lib, "nonblocking p2p send failed");
+        return status;
+    }
+
+    return status;
+}
+
+ucc_status_t ucc_tl_mlx5_mcast_p2p_recv_nb(void* src, size_t size, int rank, int tag,
```

Comment:
is tag argument needed?

Reply:
for now, we are ignoring the input tag, however, for the sake of the users of these p2p APIs, we are providing a tag field. We might redesign the p2p library to actually support the tag field.

Reply:
is this redesign going to happen soon? If not I would rather remove that argument. Otherwise please add a check that if tag is provided function returns error